yb-voyager/cmd/assessMigrationCommandBulk.go

```diff
@@ -0,0 +1,316 @@
+			if !continueOnError {
+				break
+			}
+		}
+	}
+
+	err = generateBulkAssessmentReport(dbConfigs)
+	if err != nil {
+		utils.ErrExit("failed to generate bulk assessment report: %s", err)
+	}
+}
+
+func executeAssessment(dbConfig AssessMigrationDBConfig) error {
+	log.Infof("executing assessment for schema %q", dbConfig.Schema)
+	exportDirPath := getAssessmentExportDirPath(dbConfig)
+	cmdArgs := buildCommandArguments(dbConfig, exportDirPath)
+	if err := os.MkdirAll(exportDirPath, 0755); err != nil {
+		return fmt.Errorf("creating export-directory %q for schema %q: %w", exportDirPath, dbConfig.Schema, err)
+	}
+
+	execCmd := exec.Command(os.Args[0], cmdArgs...)
+	// password either has to be provided via fleet_config_file or can be provided at run-time by the user.
+	// user setting the env var route is not supported for assess-migration-bulk command
+	execCmd.Env = append(os.Environ(), "SOURCE_DB_PASSWORD="+dbConfig.Password)
+	execCmd.Stdout = os.Stdout
+	execCmd.Stderr = os.Stderr
+	execCmd.Stdin = os.Stdin
+	log.Infof("executing the cmd: %s", execCmd.String())
+	err := execCmd.Run()
+	if err != nil {
+		return fmt.Errorf("error while assess migration of schema-%s: %v", dbConfig.Schema, err)
+	}
+	return nil
+}
+
+func buildCommandArguments(dbConfig AssessMigrationDBConfig, exportDirPath string) []string {
+	log.Infof("building assess-migration command arguments for schema %q", dbConfig.Schema)
+	args := []string{"assess-migration",
+		"--source-db-type", dbConfig.DbType,
+		"--source-db-schema", dbConfig.Schema,
+		"--export-dir", exportDirPath,
+	}
+
+	if dbConfig.User != "" {
+		args = append(args, "--source-db-user", dbConfig.User)
+	}
+	if dbConfig.TnsAlias != "" {
+		args = append(args, "--oracle-tns-alias", dbConfig.TnsAlias)
+	}
+	if dbConfig.SID != "" {
+		args = append(args, "--source-db-name", dbConfig.SID)
+	}
+	if dbConfig.Host != "" {
+		args = append(args, "--source-db-host", dbConfig.Host)
+	}
+	if dbConfig.Port != "" {
+		args = append(args, "--source-db-port", dbConfig.Port)
+	}
+	return args
+}
+
+func parseFleetConfigFile(filePath string) ([]AssessMigrationDBConfig, error) {
+	log.Infof("parsing fleet config file %q", filePath)
+	var dbConfigs []AssessMigrationDBConfig
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) == 0 {
+			continue
+		}
+		dbConfig := parseFleetConfigLine(line)
+		dbConfigs = append(dbConfigs, dbConfig)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	return dbConfigs, nil
+}
+
+/*
+Format: covers both SSL and non-SSL cases
+
+	<dbtype>,<hostname>,<port>,<service_name>,<sid>,<tns_alias>,<username>,<password>,<schema>
+*/
+func parseFleetConfigLine(line string) AssessMigrationDBConfig {
+	config := strings.Split(line, ",")
+	return AssessMigrationDBConfig{
```

Comment:
we should validate no. of arguments/length of array before processing it

Reply:
Ah I see you have a TODO 👍

Reply:
yeah will do that along with the change of header for config file.